src/locationsettings.cpp

```diff
@@ -814,3 +839,17 @@ void LocationSettingsPrivate::writeSettings()
         }
     }
 }
+
+bool LocationSettingsPrivate::isEmergencyCallActive()
+{
+    return m_callstate == QMceCallState::Active && m_calltype == QMceCallState::Emergency;
```

Comment:
Type can be "emergency" only if/when there is active (or incoming) emergency call, so in theory one could ignore state, test just the type (and get those theoretical incoming emergency calls handled for free).

Reply:
Think type as substate of ongoing call. When there is no call i.e. state=="none" , type is dontcare (but shows up as "normal"). When there is an active call, type can be "emergency" based on what ofono reports for call/modem. Technically there is nothing stopping from incoming call being labeled as emergency too - but I do not recall if that is something that can actually happen. But if it can....

Checking for "active and emergency" is true when call to emergency number is initiated / call from emergency number has been answered.

Whereas "(active or incoming) and emergency" == "emergency" is true like above + also when call from emergency number is ringing but has not been answered yet.

It is fine as it is and I guess which way to test depends on how it needs to behave  -> I'll leave it up to you whether these obscure corner case details matter or not ;-)

Reply:
Yep, thanks for clarifying this.